### PR TITLE
Adding recipe validation to modulestest

### PR DESCRIPTION
**Pull request by AhmedBM**
## Description

* Added required properties and validation to `modulestest` recipe parsing

## Checklist

- [X] I have read the [contribution guidelines](https://github.com/Azure/azure-osconfig/blob/main/CONTRIBUTING.md).
- [X] I added unit-tests to validate my changes. All unit tests are passing.
- [X] I have merged the latest `main` branch prior to this PR submission.
- [X] I submitted this PR against the `main` branch.

### CODE REVIEW

File: src/modules/test/unit-tests/TestRecipeParserTests.cpp [hunk 1]
```diff
@@ -25,4 +25,20 @@ namespace Tests
...
+    TEST(TestRecipeParserTests, RequiredFieldsMissing)
+    {
+        std::stringstream newBuffer;
+        std::streambuf *originalBuffer = std::cerr.rdbuf();
+        // Redirect cerr to our buffer
+        std::cerr.rdbuf(newBuffer.rdbuf());
+
+        TestRecipes testRecipes = TestRecipeParser::ParseTestRecipe("./recipes/testInvalid.json");
+        ASSERT_EQ(testRecipes->size(), 1);
+        EXPECT_STRCASEEQ("Test recipe './recipes/testInvalid.json' [1] missing required field: ObjectName\nTest recipe './recipes/testInvalid.json' [1] missing required field: Desired\nTest recipe './recipes/testInvalid.json' [1] missing required field: ExpectedResult\n", newBuffer.str().c_str());
```
Comment (MariusNi) [inline src/modules/test/unit-tests/TestRecipeParserTests.cpp:38]: >EXPECT_STRCASEEQ("Test r [](http://example.com/codeflow?start=8&length=24)

nit: can this super long line be broken into multiple lines?

File: src/modules/test/unit-tests/TestRecipeParserTests.cpp [hunk 2]
```diff
@@ -25,4 +25,20 @@ namespace Tests
         EXPECT_STRCASEEQ("", testRecipes->at(1).m_payload.c_str());
         EXPECT_EQ(0, testRecipes->at(1).m_payloadSizeBytes);
     }
+
+    TEST(TestRecipeParserTests, RequiredFieldsMissing)
+    {
+        std::stringstream newBuffer;
+        std::streambuf *originalBuffer = std::cerr.rdbuf();
+        // Redirect cerr to our buffer
```
Comment (MariusNi) [inline src/modules/test/unit-tests/TestRecipeParserTests.cpp:33]: >// Redirect cerr to our buffer [](http://example.com/codeflow?start=8&length=30)

Does this need to be reset once done or is self-reverting?

**Review by MariusNi — APPROVED** [review]
:shipit:

File: src/modules/test/unit-tests/TestRecipeParserTests.cpp (hunk 2 above)
Comment (AhmedBM) [reply src/modules/test/unit-tests/TestRecipeParserTests.cpp:33]: Its not self-reverting which is why we save/restore the original stream at the end

**Review by MariusNi — APPROVED** [review]
:shipit:

**Pull request merged**